chore: bump imgui, fabritone, and JVM targets

nopjmp opened the pull request:
kotlin-graphics has had a few bugs fixed that have shown up in my usage.

* Sliders would not change when manually entered a number
* Drop-down menus would not function without the layout shuffling sometimes.

Bumping the dependencies resolved the issues that I was having.

While we are here, bump Fabritone to fix the CI build.

Integration is untested, but working on some Baritone experiments.

I haven't bump this to 1.16.5 just yet even though it should just work as it's mostly server stability fixes.

nopjmp committed:
chore: bump imgui, fabritone, and JVM targets

kotlin-graphics has had a few bugs fixed that have shown up in my usage.

* Sliders would not change when manually entered a number
* Drop-down menus would not function without the layout shuffling sometimes.

Bumping the dependencies resolved the issues that I was having.

While we are here, bump Fabritone to fix the CI build.

Integration is untested, but working on some Baritone experiments.

nopjmp committed:
workflows: bump Java version to 11

5HT2 commented on build.gradle.kts:
```diff
@@ -82,7 +82,12 @@ dependencies {
     includedModImpl("net.fabricmc.fabric-api:fabric-resource-loader-v0:0.2.9+e5d3217f4e")
     includedModImpl("com.github.Ladysnake:Satin:1.5.0")
-    modImplementation("com.gitlab.CDAGaming:fabritone:fabritone~1.16.x-Fabric-SNAPSHOT")
+    modImplementation("com.gitlab.CDAGaming:fabritone:fabric~1.16.3-SNAPSHOT") {
```
you should add a comment to point out that this tag is actually 1.16.4 (if I'm not mistaken)

5HT2 commented on build.gradle.kts:
```diff
@@ -82,7 +82,12 @@ dependencies {
     includedModImpl("com.github.Ladysnake:Satin:1.5.0")
-    modImplementation("com.gitlab.CDAGaming:fabritone:fabritone~1.16.x-Fabric-SNAPSHOT")
+    modImplementation("com.gitlab.CDAGaming:fabritone:fabric~1.16.3-SNAPSHOT") {
+        exclude(group = "org.lwjgl.lwjgl")
```
formatting

5HT2 commented on build.gradle.kts:
```diff
@@ -82,7 +82,12 @@ dependencies {
-    modImplementation("com.gitlab.CDAGaming:fabritone:fabritone~1.16.x-Fabric-SNAPSHOT")
+    modImplementation("com.gitlab.CDAGaming:fabritone:fabric~1.16.3-SNAPSHOT") {
+        exclude(group = "org.lwjgl.lwjgl")
+        exclude(group= "net.java.jinput")
```
formatting :/

Ctrl alt L in intellij

nopjmp replied:
rip I forgot that committing outside of Intellij doesn't format. I'm an idiot LOL

5HT2 commented on gradle.properties:
```diff
@@ -4,20 +4,20 @@ kotlin.code.style=official
-loader_version=0.9.3+build.207
-loom_version=0.5-SNAPSHOT
+yarn_mappings=1.16.4+build.9
+loader_version=0.11.1
```
086 asked me not to bump to loader so less people are forced to update, in previous prs

nopjmp replied:
Unfortunately, we need to bump the loader from 0.9.3 to at least 0.10.1 to get the fabricmc mixin 0.8.2. I would consider dropping down to near 0.10.x We shouldn't use Mixin 0.8.1. Also, 0.10.1 allows for flexibility around dependency overwriting which may be needed by some mods if jar files conflict. It's also the recommended version from Fabric going forward, additionally includes https://github.com/FabricMC/fabric-loader/commit/5cd3b1b1bd3f12525bc5245a6edaeb657c146ebe which from what I understand will give a better message when Fabric detects a mod using newer language (like we are now Java 11).

5HT2 replied:
As long as it's not unnecessary, that's fine. Looks good to me

5HT2 replied:
And yes, Mixin 0.8.1 is a bit weird and unstable

nopjmp replied:
I'm thinking 0.10.x is safe, but I'll probably need to survey what mods out there that people use in combination.

Lithium uses 0.11.1 in dev and 0.10.6 tagged for stable. I may not know the consequences fully, so there may be a good reason.

zeroeightysix replied:
I'm confident this'll work fine - KAMI isn't using anything advanced from loader anyways.

nopjmp committed:
chore: format build.gradle.kts and comment on the version choice.

zeroeightysix commented on gradle.properties:
```diff
@@ -4,20 +4,20 @@ kotlin.code.style=official
-gli_version=4c290e0d
-gln_version=2c26a366
-uno_version=2599943c
+kg_version=7347a5d238
+unsigned_version=1e2fda82
+kool_version=b393e4c2
+glm_version=a4185eec
+gli_version=9c67885f
+gln_version=3e59f9ea
+uno_version=f60532b7
```
Did you get these from the versions imgui uses, or are they just the latest commits? If it's the latter, I would prefer you get them from the imgui repo instead.

Mind that if this is a newer major version of imgui, some testing is due.

nopjmp replied:
kg_version is the latest imgui commit

the rest are the same from the last version bump in the kg/imgui gradle.properties.

I've tested it locally, but I still want to go through everything and make sure it's working properly as I would expect from imgui proper.

I will say I noticed a lot more tests being added to kg/imgui so I'm a bit more confident that it's not going to blow up.

zeroeightysix reviewed (commented):
Users with a java 8 installation will still be able to run KAMI after this bump, right? If not, we might want to consider staying on 8.

nopjmp commented:
Yes, users will need to move to a Java 11 installation. Should I update the readme to include AdoptOpenJDK as the preferred JDK installation? It's the one I test with and use primarily. Oracle's official Java 11 would work as well, but I refuse to install that due to personal/work/ethical obligations. Other JDKs should work fine.

The only way we could stay targeting Java 8 is to either move off of kg/imgui or use the older release version + a few commits before they forced Java 11.

Alternatively, we could work with kg/imgui to see if we can keep Java 8 compatibility and remove the Java 11 requirement.

zeroeightysix commented:
> Yes, users will need to move to a Java 11 installation. Should I update the readme to include AdoptOpenJDK as the preferred JDK installation? It's the one I test with and use primarily. Oracle's official Java 11 would work as well, but I refuse to install that due to personal/work/ethical obligations. Other JDKs should work fine.

This should be mentioned in the README. I'd prefer the JRE that is easiest to install to be linked. 
> The only way we could stay targeting Java 8 is to either move off of kg/imgui or use the older release version + a few commits before they forced Java 11.
 
I don't know what the java 11 coverage is on e.g. windows. What percentage of normal users have 11 installed and would be able to run kami as is, without installing anything extra? If it's a reasonable amount, I don't mind moving to 11.
> Alternatively, we could work with kg/imgui to see if we can keep Java 8 compatibility and remove the Java 11 requirement.

This is the reason the [jdk8](https://github.com/kotlin-graphics/imgui/tree/jdk8) branch exists. If you open an issue asking for the branch to be updated (or, even better, a PR), I'm sure elect would get it done quickly.

5HT2 commented:
> I'd prefer the JRE that is easiest to install to be linked.

Openjdk is in basically all mainstream distro repos, and for non-Linux the installation is basically point and click.

nopjmp commented:
> This is the reason the [jdk8](https://github.com/kotlin-graphics/imgui/tree/jdk8) branch exists. If you open an issue asking for the branch to be updated (or, even better, a PR), I'm sure elect would get it done quickly.

Posted a question regarding testing 1.80 for JDK8 so we'll see what happens. If I find time this evening to make a PR/fork. If needed, we can always just have a separate fork to maintain compatibility to what we want and just pull in from that. It would help keep the development loop smaller and then contribute fixes back upstream.

SpaiR commented:
Hi. 
My post may be a bit out of context, but seeing your problem I would recommend  to consider to move to the native bindings instead. It uses the original Dear ImGui lib and has other good points think about. https://github.com/SpaiR/imgui-java

5HT2 commented:
I'm not sure how Java bindings would be better over Kotlin bindings, and the description says it's based off of Dear ImGui so it would most likely have the same problems here

nopjmp commented:
> I'm not sure how Java bindings would be better over Kotlin bindings, and the description says it's based off of Dear ImGui so it would most likely have the same problems here

kg/imgui is a kotlin reimplementation.

Additionally, I would rather avoid native-bindings as that would require more work to support new platforms. Currently, lwjgl is the only native dependency requires porting to different architectures/platforms (i.e. new Apple Silicon Macs even though they do have x86 emulation or NixOS and their weird user land setup).

SpaiR commented:
> over Kotlin bindings

That is the point. kotlin-graphics is not a binding. It is the port. It doesn't uses the original library. It reimplements it. And issues you've encountered are the issues of the port, not the original library. From what is on the surface, binding doesn't care about java version. It's much smaller, has no a ton of dependecies and has the latest features which just work. It also can be used as a binding for other Dear ImGui extensins like this PR does https://github.com/SpaiR/imgui-java/pull/29

> I would rather avoid native-bindings as that would require more work to support new platforms.

Yet you are willing to create a fork of kotlin implementation and maintain its compatibility... Binding has builts for all major platforms, with no other dependecies. The only one is for LWJGL by the renderer and GLFW backend, which is optional as well.

Don't insist on anything or argue about kotlin variant advantages. Just making the point clear. If you're fine with current solution, then np.

zeroeightysix committed:
Disable imgui debug mode

For some reason, this now defaults to true. I guess elect accidentally pushed this while debugging something. Luckily we can just turn it off on init. Leaving it on clutters the log with spammy debug messages.

zeroeightysix committed:
Replaced usages of %s formatting in imgui calls

For some reason, this was un-implemented? It didn't work very well before either way. Its usages have been replaced.

zeroeightysix commented:
![image](https://user-images.githubusercontent.com/27009727/105640857-bb87dc00-5e80-11eb-88c4-2d57caea122a.png)
Last problem before merging is the fact that the default font no longer has a name? I can try delving into why soon. If you do it before me, that's fine too.

nopjmp commented:
I'm actually considering rolling into the 1.80 branch instead of master, it looks like some fixes were directly pushed there.

zeroeightysix commented:
Considering #520 fixes the above problem, I'm going to merge this for now as it's really minor anyways.

I think having the build work again is worth it.

I'd like to thank @SpaiR for his input, actually: the bindings are working better than I expected.
